src/lib/profiles/data-management/Current/LoggingManagement.cpp

```diff
@@ -2117,7 +2121,9 @@ void CircularEventBuffer::AddEventUTC(utc_timestamp_t inEventTimestamp)
 
 void CircularEventBuffer::RemoveEvent(size_t aNumEvents)
 {
+    event_id_t currentFirstEventID = mFirstEventID;
     mFirstEventID += aNumEvents;
+    WeaveLogProgress(EventLogging, "Dropping events | Move first event id from %u to %u", currentFirstEventID, mFirstEventID);
```

Comment:
I think with this log line in place you're looking at a whole lot of log lines.  In a steady state, you're probably looking at (an average) of one of the above lines per LogEvent.  This is no longer a `Progress` level of logging, more like `debug`. Because `WeaveLog*` family of functions are synchronous, these log lines can occupy a significant time to output, particularly when the underlying serial console is slow.  `WeaveLogDebug` would be far more appropriate here.

Reply:
I thought RemoveEvent is only called here https://source.corp.google.com/nest/platform/weave/master/src/lib/profiles/data-management/Current/LoggingManagement.cpp;rcl=6854575c8beb472be83b67590f9830730b733487;l=1781
when events are being dropped due to overflow. This is not where we move the counter when fetching events. And since the buffer size are increased, this shouldn't happen that often. Is my understanding wrong?

Reply:
In steady state -- all event buffers are full and the average size of event is stationary -- we expect that for every event logged, we need to remove one to free the space in the circular buffers.

Reply:
you mean when we build the event notify, we don't change mFirstEventID? So the events removed here, some of them actually have been sent to service?